### ISSUE TITLE
Mot de passe oublié ?

### ISSUE THREAD

**Issue by Pixflowave** [l-atelier-des-chercheurs/dodoc#512]
...
est-il possible d'avoir la fonction Mot de passe oublié ?

**Comment by louis-ev** [comment]
Qui ferait quoi ? Car ajouter un mail sur un auteur n’est pas obligatoire.
On peut imaginer qu’un admin peut attribuer un nouveau mot de passe éventuellement.

**Comment by louis-ev** [comment]
Maintenant que ça c’est implémenté, question suivante : comment fait une personne qui aurait oublié son mot de passe ? Y a t’il une indication qqpart pour contacter un admin pour qu’il attribue un nouveau mdp ? "Vous avez oublié votre mot de passe ? Cliquez ici pour envoyer un email à un administrateur pour lui demander de vous créer un nouveau mot de passe."

**Comment by PixnWave** [comment]
mots de passe oublié -> mail admin
où boucle de modification de mot de passe autonome...

**Comment by sarahgarcin** [comment]
le mail admin me parait le plus logique, ça me semble un ajout important

**Comment by PixnWave** [comment]
Perso je préférerais le boucle autonome pour ne pas avoir à un intervenir dans ce processus le mail Admin viendrait en dernière recours...

**Comment by louis-ev** [comment]
Quand tu parles de "boucle de modification de mot de passe autonome" tu veux dire un bouton "mot de passe oublié" ? Avec envoie de mail si l’utilisateur a renseigné son mot de passe ?

**Comment by PixnWave** [comment]
- Compte avec mail 
Bouton envoyé un mail pour réinitialisé le mot de passe
- compte sans mail
Bouton envoyé un mail pour admin

**Comment by jubonhomme** [comment]
La solution du bouton "j'ai oublié mon mot de passe " suivit d'un mail automatique me semble la plus intéressante pour l'administrateur.
Il n'a plus a intervenir à chaque fois et n'a pas connaissance / accès aux mots de passe de chaque utilisateur, ça me parait plus conforme niveau légal aussi.
Mais c'est possible que si le champ mail est renseigné

Dans les cas d'une classe avec élèves ça ne fonctionne pas puisque pas de mail.... 
Et dans le cas d'un compte sans mail comme le propose @PixnWave (non renseigné ET / OU pas de mail existant), l'administrateur reçoit une demande de réinitialisation... mais comment l'administrateur contacte l'utilisateur pour lui transmettre le nouveau mot de passe ?

Retour case départ... il faut un moyen de contacter l'auteur en dehors de dodoc (par mail ou en présence)

**Comment by jubonhomme** [comment]
Je vois deux types de fonctionnement différents : 

- pour un dodoc local sur un poste dans une classe par exemple : pas de mail obligatoire / réinitialisation par l'administrateur si besoin par le dossier dodoc2 pour les quelques utilisateurs... pas de grosse charge 

- pour un dodoc serveur : beaucoup plus d'utilisateurs -> besoin de passer par une solution automatisée de récupération de mot de passe par mail -> l'auteur n'est pas juste un tag mais un vrai compte avec identification et contact soit par email soit fédération d'identité (compte académique / ENT / AppsEdu...)

**Comment by louis-ev** [comment]
>  Edit : Je viens de voir que le mot de passe n'est pas en clair dans le dossier dodoc2, l'administrateur n'y a donc pas accès directement ? Il peut simplement le supprimer ?

Exact. Il est chiffré. L’admin peut simplement retirer le mot de passe d’un auteur, ou lui en attribuer un nouveau.

> pour un dodoc serveur : beaucoup plus d'utilisateurs -> besoin de passer par une solution automatisée de récupération de mot de passe par mail -> l'auteur n'est pas juste un tag mais un vrai compte avec identification et contact soit par email soit fédération d'identité (compte académique / ENT / AppsEdu...)

Je me demandais si on devrait pas considérer que si un utilisateur veut protéger son compte avec un mot de passe, il doit entrer un mail pour pouvoir en récupérer un nouveau s’il en fait la demande. Mais le soucis c’est la situation locale comme tu l’évoques.

On pourrait avoir un message qui apparaît si tu mets un mot de passe avant la validation du compte : *Attention ! Si vous ne remplissez pas le champ "mail", il vous faudra contacter un administrateur si vous oubliez votre mode passe.* ?

**Comment by jubonhomme** [comment]
C'est en place sur la V12... une très bonne nouvelle !

https://framateam.org/latelier-des-chercheurs/pl/jab3h79s1pbit8cycibmcugdfe

A fermer ?
